common/lib/xmodule/xmodule/mongo_utils.py

```diff
@@ -30,10 +30,18 @@ def connect_to_mongodb(
     handles AutoReconnect errors by retrying read operations, since these exceptions
     typically indicate a temporary step-down condition for MongoDB.
     """
+    # Appsembler Specific: This code allow us to have a default replica set value
+    # in our configuration playbooks.
+    if kwargs.get('replicaSet', '') == '':
+        # If the replicaSet is empty let's set it as None, that is the default
+        # value that pymongo has. If not, will look for a '' replica set.
+        kwargs.get('replicaSet') = None
     # The MongoReplicaSetClient class is deprecated in Mongo 3.x, in favor of using
     # the MongoClient class for all connections. Update/simplify this code when using
     # PyMongo 3.x.
-    if kwargs.get('replicaSet'):
+    if kwargs.get('replicaSet') and kwargs.get('replicaSet') is not None:
```

Comment:
That should be enough, `None` is falsy.

```suggestion
    if kwargs.get('replicaSet'):
```

Please do what works and let's refactor tomorrow or anytime soon.